compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -1977,6 +1977,9 @@ impl<'a> Parser<'a> {
         open_delim_span: Span,
     ) -> PResult<'a, ()> {
         if self.token.kind == token::Comma {
+            if !self.sess.source_map().is_multiline(prev_span.until(self.token.span)) {
```

Comment:
This refers to `Parser::expected_one_of_not_found`.

https://github.com/rust-lang/rust/blob/380addd7d2971447d7f6828c508a93fa8018a9b6/compiler/rustc_parse/src/parser/diagnostics.rs#L905-L906